test/types/animation.ts

```diff
@@ -20,7 +20,8 @@ const chart = new Chart('id', {
         delay: (ctx) => ctx.dataIndex * 100,
         duration: (ctx) => ctx.datasetIndex * 1000,
         loop: true,
-        easing: 'linear'
+        easing: 'linear',
+        fn: (from: any, to: any, factor: number) => ({size: Animation.interpolators.number(0, 32, factor)} as any)
```

Comment:
this is a really bad test, animating numbers a, and b and resulting to an object value.

I would argue it would be better to fix the options resolution for `font` in the use case and just animate the size as a number.. and not support animation objects, because its going to open deep can of worms.

Reply:
That said, I'm not against this PR itself. I'd even maybe make the interpolators writable, if that can be supported.

Reply:
@kurkle when you wrote ‘writable’ do you mean custom interpolator for custom type?

Reply:
> this is a really bad test, animating numbers a, and b and resulting to an object value.

@kurkle this is just a test for the OOTB interpolators. I can add a specific test for objects but I didn't want to add another test case because fn was used here.

> I would argue it would be better to fix the options resolution for font in the use case and just animate the size as a number.. and not support animation objects, because its going to open deep can of worms.

@kurkle I could agree with you but, afai understood, `fn` option was born for this reason, to give to the user the possibility to animate items not manageable OOTB (like objects). If there is this option for other purposes, it's not clear to me.
About font, this is not really my use case. The use case is coming from annotation plugin where, having color option aa an array, the animation doesn't recognize the array, therefore I was thinking to have a specific animation for color where, when configured as an array, the custom interpolator will return an array of intepolated colors.

The best solution could be to enable animation on inner nodes of the options (like font, padding, borders and other), where the property is defined by dot separator. Something like:

```javascript
properties: ['font.size', 'padding.top'],
```

But I think this deserve a specific PR, this one is just to expose the OOTB interpolator.
Let me know how to proceed.